lib/proto.js

```diff
@@ -165,6 +170,9 @@ function write (_obj, msg, num) {
   const stream = this[streamSym]
   if (stream[needsMetadataGsym] === true) {
     stream.lastLevel = num
+    if (msg) {
+      obj[this[messageKeySym]] = msg
+    }
```

Comment:
Why is this here?

Reply:
If we only pass the obj, if the `nestedKey` option is activated then the obj will have no msg key. If the option is false, then the obj will have the key.

I did this to be as close as possible to the  `nestedKey=false` behavior.

Reply:
The `msg` property is no longer placed on the `obj` in the `asJson` function.

Thing is, the `lastMsg` is also placed here so this is not requeired, but the behaviour will be a little diffrent (although arguably more correct), which I tried to avoid as much as possible, while keeping things performant.

Reply:
After reading #882 I realised the `msg`, as deduced by the algorithm, maybe should not be in the `obj` at all.

This is why you guys put in the `originalMsg`, because you basically wanted both messages. But this is why I used the `nestedKey` (to avoid all collisions).

So maybe the good fix is to not replace the `msg` property in the `obj` and use `nestedKey`.

The [new handling of the `msg` algorithm in the PR](https://github.com/mihai1voicescu/pino/blob/783bbfa20db90a8a617057242851b60c265ee361/lib/tools.js#L146) will place the `msg` key as the last thing in the JSON log w/o interfering with the `obj`.

Reply:
I will remove the if, and leave the obj without the msg. This is available in lastMsg